modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableColumnHeaderTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene.control.skin;
+
+import com.sun.javafx.tk.Toolkit;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.event.EventType;
+import javafx.scene.Scene;
+import javafx.scene.control.Skin;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableColumnBase;
+import javafx.scene.control.TablePosition;
+import javafx.scene.control.TableView;
+import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.scene.control.skin.NestedTableColumnHeader;
+import javafx.scene.control.skin.TableColumnHeader;
+import javafx.scene.control.skin.TableHeaderRow;
+import javafx.scene.control.skin.TableViewSkin;
+import javafx.stage.Stage;
+import org.junit.Before;
+import org.junit.Test;
+import test.com.sun.javafx.scene.control.test.Person;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class TableColumnHeaderTest {
+
+    private MyTableColumnHeader tableColumnHeader;
+
+    @Before
+    public void beforeTest() {
+        tableColumnHeader = null;
+    }
+
+    @Test
+    public void test_resizeColumnToFitContent() {
```

Comment:
This method actually is three-in-one :) It's testing

a) trigger a resize initially doesn't change the with 
b) change content to larger and trigger a resize increases column width
c) change content to smaller and trigger a resize decreases column width

Not sure about conventions here, but I would separate them out into distinct methods.

Reply:
Well that depends on the point of view. From my point of view, I'm testing the "resize to fit content" feature. So I'm testing one feature and therefore the same size, a bit smaller and a bit greater. 
But we could split this into 3 separate tests if that is the convention.

Reply:
so you end up with one test per public/protected method? Hmm ..